tika-parsers/src/main/java/org/apache/tika/parser/recognition/ObjectRecognitionParser.java

```diff
@@ -117,55 +130,68 @@ public synchronized void parse(InputStream stream, ContentHandler handler, Metad
         }
         metadata.set(MD_REC_IMPL_KEY, recogniser.getClass().getName());
         long start = System.currentTimeMillis();
-        List<RecognisedObject> objects = recogniser.recognise(stream, handler, metadata, context);
+        List<? extends RecognisedObject> objects = recogniser.recognise(stream, handler, metadata, context);
+
         LOG.debug("Found {} objects", objects != null ? objects.size() : 0);
         LOG.debug("Time taken {}ms", System.currentTimeMillis() - start);
+
         if (objects != null && !objects.isEmpty()) {
+            int count;
+            List<RecognisedObject> acceptedObjects = new ArrayList<RecognisedObject>();
+            List<String> xhtmlIds = new ArrayList<String>();
+            String xhtmlStartVal = null;
+
+            if (recogniser instanceof TensorflowRESTRecogniser || recogniser instanceof TensorflowImageRecParser) {
```

Comment:
:-1:  There is a better way to handle this.

In Model and Services terminalogy, we have
 `TensorflowRESTRecogniser`, `TensorflowImageRecParser`, and `TensorflowRESTCaptioner` as services
`RecognisedObject` and `CaptionObject` as models.

The problem:
the condition is on the service, i.e. `recogniser instanceof TensorflowRESTRecogniser`
What if we add a new awesome service tomorrow? We need to change this code right?

The solution:
  Make your decision based on the model object

i.e. check 
```java
for (RecognisedObject object: objects) {
  if (object instanceof CaptionObject`) {
    This result is a caption
  } else {
    this is from something else, the default one
  }
}
```
As long as the new services return data in the same model, the code will work.
Let me know if this needs more explanation!